packages/block-editor/src/components/link-control/index.js

```diff
@@ -210,8 +210,7 @@ function LinkControl( {
 		// is still unassigned after calling `onChange`.
 		const hadFocusLoss =
 			isEndingEditWithFocus.current &&
-			wrapperNode.current &&
-			! wrapperNode.current.contains( document.activeElement );
+			! wrapperNode.current?.contains( document.activeElement );
```

Comment:
This line of code is why everything started. After #19831, everyone thought that we could use optional chaining in Gutenberg. 

But we couldn't because `espree` 4.x in docgen cannot parse it. @aduth thought we can simply do it by updating `espree` to the latest version. Unfortunately, espree doesn't support `leadingComment` properties after 5.0. So, we had to change the parser. (Older version of docgen extract JSDoc from `leadingComment` and parsed it with `doctrine`. And `doctrine` is deprecated and unused.)

As TypeScript can parse JS and JSDoc, it was chosen. 

This line means that we can use optional chaining without any error.

Reply:
I don't think we need to back it out at this point, but I'd generally suggest to find ways to limit the scope of a pull request as much as possible, especially if it's already a +3400/-8400 proposal. It's definitely related as the original motivation of the changes, but I think it could have been fine to do as a separate follow-up pull request.

Reply:
I reverted it and added new test `new-syntax` under `test/fixtures`. Later, when new syntax doesn't work or want to check if a new syntax works, we can use this file.